editor/src/components/navigator/navigator-item/navigator-item-wrapper.tsx

```diff
@@ -72,72 +84,145 @@ const labelSelector = createCachedSelector(
     }
     return MetadataUtils.getElementLabelFromMetadata(allElementProps, elementMetadata)
   },
-)((_, targetPath) => EP.toString(targetPath))
+)((_, navigatorEntry) => navigatorEntryToKey(navigatorEntry))
 
 const elementWarningsSelector = createCachedSelector(
   (store: DerivedSubstate) => store.derived.elementWarnings,
-  (_: DerivedSubstate, elementPath: ElementPath) => elementPath,
-  (elementWarnings, elementPath) => {
-    return (
-      getValueFromComplexMap(EP.toString, elementWarnings, elementPath) ?? defaultElementWarnings
-    )
+  (_: DerivedSubstate, navigatorEntry: NavigatorEntry) => navigatorEntry,
+  (elementWarnings, navigatorEntry) => {
+    if (isRegularNavigatorEntry(navigatorEntry)) {
+      return (
+        getValueFromComplexMap(EP.toString, elementWarnings, navigatorEntry.elementPath) ??
+        defaultElementWarnings
+      )
+    } else {
+      return defaultElementWarnings
+    }
   },
-)((_, targetPath) => EP.toString(targetPath))
+)((_, navigatorEntry) => navigatorEntryToKey(navigatorEntry))
 
 const noOfChildrenSelector = createCachedSelector(
   (store: DerivedSubstate) => store.derived.navigatorTargets,
-  (_: DerivedSubstate, targetPath: ElementPath) => targetPath,
-  (navigatorTargets, targetPath) => {
+  (_: DerivedSubstate, navigatorEntry: NavigatorEntry) => navigatorEntry,
+  (navigatorTargets, navigatorEntry) => {
     let result = 0
     for (const nt of navigatorTargets) {
-      if (EP.isChildOf(nt, targetPath)) {
+      if (
+        isRegularNavigatorEntry(navigatorEntry) &&
+        EP.isChildOf(nt.elementPath, navigatorEntry.elementPath)
+      ) {
         result += 1
       }
     }
     return result
   },
-)((_, targetPath) => EP.toString(targetPath))
+)((_, navigatorEntry) => navigatorEntryToKey(navigatorEntry))
+
+function getNavigatorEntryLabel(
+  navigatorEntry: NavigatorEntry,
+  labelForTheElement: string,
+): string {
+  switch (navigatorEntry.type) {
+    case 'REGULAR':
+      return labelForTheElement
+    case 'CONDITIONAL_CLAUSE':
+      switch (navigatorEntry.clause) {
+        case 'then':
+          return 'Then'
+        case 'else':
+          return 'Else'
+        default:
+          throw assertNever(navigatorEntry.clause)
+      }
+    case 'SYNTHETIC': {
+      if (childOrBlockIsChild(navigatorEntry.childOrAttribute)) {
+        switch (navigatorEntry.childOrAttribute.type) {
+          case 'JSX_ELEMENT':
+            return getJSXElementNameLastPart(navigatorEntry.childOrAttribute.name)
+          case 'JSX_ARBITRARY_BLOCK':
+            return 'Unknown'
```

Comment:
I _think_ this should either be `Code` or the actual code itself (which is what we do right now)

Reply:
Addressed in this PR: https://github.com/concrete-utopia/utopia/pull/3399